src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/UpdateOperation.java

```diff
@@ -50,6 +50,13 @@ public boolean shouldApplyIf(boolean isInsert) {
    * specifically Document's primary id, {@code _id}.
    */
   protected static String validateUpdatePath(UpdateOperator oper, String path) {
+    if (DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD.equals(path)
+        && !(oper.operator().equals("$set") || oper.operator().equals("$unset"))) {
```

Comment:
Can we also support `$setOnInsert` for `$vector`? Not a huge deal, but I don't see any reason why we can't support `$setOnInsert`.

Reply:
OK will add $setOnInsert operation for vector search